### PR TITLE
Doxygen for gbytec, gbytes, gbytesc, getbit, getgb (Part of #77)

### PR DESCRIPTION
**Pull request by BrianCurtis-NOAA**
Part of #77 

Doxygen for:
- gbytec
- gbytes
- gbytesc
- getbit
- getgb

### CODE REVIEW

**Comment by BrianCurtis-NOAA** [comment]
Once you merge my other PR #88 it will show only 5 files changed.

File: src/gbytec.f [hunk 1]
```diff
@@ -1,4 +1,14 @@
 C> @file
+C> @brief Wrapper for gbytesc() limiting NSKIP and N to 0 and 1
```
Comment (edwardhartnett) [inline src/gbytec.f]: Punctuation...

File: src/gbytec.f [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+C> @author Unknown
```
Comment (edwardhartnett) [inline src/gbytec.f]: Change this to "NOAA Programmer" instead of "Unknown".

File: src/gbytec.f [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+
+C> Wrapper for gbytesc() limiting NSKIP and N to 0 and 1
```
Comment (edwardhartnett) [inline src/gbytec.f]: Punctuation.

File: src/gbytes.f [hunk 4]
```diff
@@ -1,41 +1,26 @@
 C> @file
-C> THIS IS THE FORTRAN VERSION OF GBYTES.
-C
-C> THIS IS THE FORTRAN VERSION OF GBYTES.
-C> @author DR. ROBERT C. GAMMILL, CONSULTANT, NATIONAL CENTER FOR ATMOSPHERIC RESEARCH
-C> @date MAY 1972
-C>
-C> CHANGES FOR SiliconGraphics IRIS-4D/25
+C> @brief This is the fortran version of gbytes.
+C> @author Dr. Robert C. Gammill @date 1972-05-DD
```
Comment (edwardhartnett) [inline src/gbytes.f]: Change date to just May, 1972.

Wow this code is 50 years old!

File: src/gbytes.f [hunk 5]
```diff
@@ -76,43 +61,43 @@ SUBROUTINE GBYTES(IPACKD,IUNPKD,NOFF,NBITS,ISKIP,ITER)
...
 C
-      IBITS  = MOD(ISTEP,NBITSW) 
+      IBITS  = MOD(ISTEP,NBITSW)
 C
       DO 10 I = 1,ITER
 C
-C MOVER SPECIFIES HOW FAR TO THE RIGHT A BYTE MUST BE MOVED IN ORDER            
+C MOVER SPECIFIES HOW FAR TO THE RIGHT A BYTE MUST BE MOVED IN ORDER
 C
-C    TO BE RIGHT ADJUSTED.
```
Comment (edwardhartnett) [inline src/gbytes.f:93]: Are you using some tool to remove whitespace at the end of the lines?

If so, in future, do a pass with just that change, and put up a PR, and then do the doxygen in a separate PR. That makes it easier to review.

File: src/gbytesc.f [hunk 6]
```diff
@@ -1,15 +1,16 @@
 C> @file
-C
-C> Get bytes - unpack bits:  Extract arbitrary size values from a
-C> packed bit string, right justifying each value in the unpacked
-C> array. v1.1
+C> @brief Get bytes - unpack bits
+C> @author Unknown
+
+C> Extract arbitrary size values from a packed bit string,
+C> right justifying each value in the unpacked array. v1.1
```
Comment (edwardhartnett) [inline src/gbytesc.f]: Remove the "v1.1".

File: src/getgb.f [hunk 7]
```diff
@@ -1,185 +1,165 @@
...
+C>  - (15) Time range 2
+C>  - (16) Time range flag
+C>  - (17) Number included in average
+C>  - (18) Version nr of grib specification
+C>  - (19) Version nr of parameter table
+C>  - (20) Nr missing from average/accumulation
+C>  - (21) Century of reference time of data
+C>  - (22) Units decimal scale factor
+C>  - (23) Subcenter number
+C>  - (24) Pds byte 29, for nmc ensemble products
```
Comment (edwardhartnett) [inline src/getgb.f]: Remove the parens around all these numbers.

File: src/getgb.f [hunk 8]
```diff
@@ -1,185 +1,165 @@
...
+C>  - (6) Resolution flag  (right adj copy of octet 17)
+C>  - (7) lov grid orientation
+C>  - (8) dx - x direction increment
+C>  - (9) dy - y direction increment
+C>  - (10) Projection center flag
+C>  - (11) Scanning mode (right adj copy of octet 28)
+C> - Spherical harmonic coefficients
+C>  - (2) j pentagonal resolution parameter
+C>  - (3) k      "          "         "
+C>  - (4) m      "          "         "
```
Comment (edwardhartnett) [inline src/getgb.f]: Replace the " with the words. Also take out the parens around the numbers. So:

C>  - 2 j pentagonal resolution parameter
C>  - 3 k pentagonal resolution parameter
C>  - 4 m pentagonal resolution parameter

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/gbytec.f (hunk 1 above)
Comment (BrianCurtis-NOAA) [reply src/gbytec.f]: NSKIP and N are variables in gbytesc() should variable names all be lowercase?

File: src/gbytes.f (hunk 5 above)
Comment (BrianCurtis-NOAA) [reply src/gbytes.f:93]: Do you want me to focus on all comments and not just the doxygen comments? Should they also be lowercase?

File: src/gbytec.f (hunk 1 above)
Comment (edwardhartnett) [reply src/gbytec.f]: No it should have a period at end of sentence. ;-)

File: src/gbytes.f (hunk 5 above)
Comment (BrianCurtis-NOAA) [reply src/gbytes.f:93]: Oh i see. I don't recall a tool being used, but there shouldn't be whitespace there.

Comment (BrianCurtis-NOAA) [reply src/gbytes.f:93]: I'll keep an eye on it. I think that was the only file edited like that.

Comment (edwardhartnett) [reply src/gbytes.f:93]: For the doxygen work, just change the comment blocks at the top of the file and above each subprogram. Don't change the comments within the code itself.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**